### PR TITLE
resrc: Add resource API context support

### PR DESCRIPTION
**Pull request by dongahn**
Add resource API context support + some minor cleanup

### CODE REVIEW

File: resrc/test/tresrc.c [hunk 1]
```diff
@@ -417,91 +426,132 @@ static int test_a_resrc (resrc_t *resrc, bool rdl)
...
+        goto ret;
+
     ok ((hwloc_topology_init (&topology) == 0),
         "hwloc topology init succeeded");
     ok ((hwloc_topology_load (topology) == 0),
         "hwloc topology load succeeded");
-    ok (((resrc = resrc_create_cluster ("cluster")) != 0),
-        "cluster resource creation succeeded");
-    ok ((resrc_generate_hwloc_resources (resrc, topology, NULL, NULL) != 0),
+    /* Generate teh hardware hiearchy of resources using hwloc reader */
```
Comment (lipari) [inline resrc/test/tresrc.c]: teh -> the

Comment (chu11) [inline resrc/test/tresrc.c]: also hiearchy -> hierarchy :-)

**Comment by lipari** [comment]
@dongahn, looks great!  I'm ready to merge once typos are fixed.  There are 4 misspelled "hiearchy" in tresrc.c.

**Comment by dongahn** [comment]
Got it. Will work on this and push.

**Comment by dongahn** [comment]
Ok. Ready.

**Pull request merged**